### PR TITLE
Feature/implementing new interface and adding  transaction methods to that interface

### PR DESCRIPTION
**Pull request by bokimilinkovic**

### CODE REVIEW

File: interfaces/interfaces.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package interfaces
```
Comment (sfqi) [inline interfaces/interfaces.go]: how about we put this in `interactor/store.go`? I'm not sure about this, but let's see what happens

File: repository/postgres/postgres.go [hunk 2]
```diff
@@ -120,3 +121,16 @@ func (store *Store) FindUserByID(id int) (*model.User, error) {
...
 }
+
+func (store *Store) Transaction() i.Store {
+	tx := store.db.Begin()
+	return NewStore(tx)
+}
+
+func (store *Store) Commit() {
+	store.db.Commit()
+}
```
Comment (sfqi) [inline repository/postgres/postgres.go]: see this for error handling http://jinzhu.me/gorm/advanced.html#a-specific-example

rollback can error as well, but let's skip that for now

File: repository/mock/store.go [hunk 3]
```diff
@@ -89,3 +90,19 @@ func (s *Store) CreateLoan(loan *model.Loan) error {
...
+func (s *Store) Transaction() interfaces.Store {
+	args := s.Called()
+	if args.Get(0) != nil {
+		return args.Get(0).(interfaces.Store)
+	}
+	return nil
+}
+
+func (s *Store) Commit() {
+	_ = s.Called()
```
Comment (sfqi) [inline repository/mock/store.go]: `s.Called()` seems simpler

File: interactor/loan.go [hunk 4]
```diff
@@ -2,20 +2,21 @@ package interactor
...
-	FindAllLoans() ([]*model.Loan, error)
-	FindLoansByBookID(int) ([]*model.Loan, error)
-	FindLoansByUserID(int) ([]*model.Loan, error)
-}
+//type loanStore interface {
+//	FindLoanByID(int) (*model.Loan, error)
+//	FindAllLoans() ([]*model.Loan, error)
+//	FindLoansByBookID(int) ([]*model.Loan, error)
+//	FindLoansByUserID(int) ([]*model.Loan, error)
+//}
```
Comment (sfqi) [inline interactor/loan.go]: no need for this

**Comment by sfqi** [comment]
I thought I said to do this in a branch made from master

File: interfaces/interfaces.go (hunk 1 above)
Comment (bokimilinkovic) [reply interfaces/interfaces.go]: it seems also fine

**Comment by bokimilinkovic** [comment]
@sfqi i am more confident that everything is working fine if I can test all interactors and handler

**Pull request merged**